src/main/kotlin/org/rust/lang/core/psi/RsPsiFactory.kt

```diff
@@ -172,6 +172,18 @@ class RsPsiFactory(private val project: Project) {
         createFromText<RsFunction>("unsafe fn foo(){}")?.unsafe
             ?: error("Failed to create unsafe element")
 
+    fun createFunction(name: String, stmts: List<PsiElement>, public: Boolean): RsFunction =
+        createFromText<RsFunction>("${if (public) "pub" else "" } fn $name() {\n${stmts.joinToString( separator = "\n", transform = { it.text })}\n}")
+            ?: error("Failed to create function element")
```

Comment:
It may be useful to include function text in the error message. If somebody reports a bug with the exception, it would be much easier to reproduce!